beacon-chain/blockchain/process_block.go

```diff
@@ -104,6 +108,9 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 	// Updating next slot state cache can happen in the background. It shouldn't block rest of the process.
 	if featureconfig.Get().EnableNextSlotStateCache {
 		go func() {
+			// Use a custom deadline here, since this method runs asynchronously.
+			ctx, cancel := context.WithTimeout(context.Background(), slotDeadline)
```

Comment:
Why does this use context.Background() instead of the parent context?

Reply:
Oh, I see in the PR description. Could you please add more context in the code commentary please?